app/Controller/DirectoryController.php

```diff
@@ -164,7 +174,14 @@ public function checkout(Request $request, Response $response, array $arguments)
         $localBranchName = substr($remoteBranchName, strpos($remoteBranchName, '/') + 1);
         $currentBranch = $gitRepository->getCurrentBranch();
         if ($currentBranch !== $localBranchName) {
-            $gitRepository->checkout(['track', ['B' => $localBranchName]], [$remoteBranchName]);
+            if (strpos($remoteBranchName, GitRepository::TAG_ORIGIN.'/') !== false) {
+                // tags
+                $gitRepository->checkout([['B' => $localBranchName]], [$remoteBranchName]);
+            } else {
+                // branches
+                $gitRepository->checkout(['track', ['B' => $localBranchName]], [$remoteBranchName]);
+            }
+
```

Comment:
It is more common to prefix branches with "dev-" instead of prefixing the tags. Would you mind to change it. Furthermore I don't see any need to add a (foreign) constant to handle this. If you want, you can use a private property for the branch prefix.